janus/lib/models/token.rb

```diff
@@ -72,10 +72,8 @@ def create_task_token!(project_name, read_only: false)
       # degrade admin permissions
       if payload[:perm] =~ /^[Aa]/
         payload[:perm] = payload[:perm].sub(/^[Aa]/) { |c| c == 'A' ? 'E' : 'e' }
-      end
-
-      # permit supereditor
-      if @janus_user.supereditor?
+        # permit supereditor
+      elsif @janus_user.supereditor?
```

Comment:
So I tried the fix `payload[:perm] ||= "e:#{project_name}"` but it causes tests to fail with 401 status:

```
 1) Token Generation task tokens allows supereditors create a task token without project permission
     Failure/Error: expect(last_response.status).to eq(200)

       expected: 200
            got: 401

       (compared using ==)
     # ./spec/token_spec.rb:246:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:170:in `block (3 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.5.0/gems/database_cleaner-1.8.5/adapters/database_cleaner-sequel/lib/database_cleaner/sequel/transaction.rb:36:in `block in cleaning'
     # ./vendor/bundle/ruby/2.5.0/gems/sequel-5.43.0/lib/sequel/database/transactions.rb:251:in `_transaction'
     # ./vendor/bundle/ruby/2.5.0/gems/sequel-5.43.0/lib/sequel/database/transactions.rb:233:in `block in transaction'
     # ./vendor/bundle/ruby/2.5.0/gems/sequel-5.43.0/lib/sequel/connection_pool/threaded.rb:92:in `hold'
     # ./vendor/bundle/ruby/2.5.0/gems/sequel-5.43.0/lib/sequel/database/connecting.rb:269:in `synchronize'
     # ./vendor/bundle/ruby/2.5.0/gems/sequel-5.43.0/lib/sequel/database/transactions.rb:195:in `transaction'
     # ./vendor/bundle/ruby/2.5.0/gems/database_cleaner-1.8.5/adapters/database_cleaner-sequel/lib/database_cleaner/sequel/transaction.rb:36:in `cleaning'
     # ./vendor/bundle/ruby/2.5.0/gems/database_cleaner-1.8.5/lib/database_cleaner/configuration.rb:87:in `block (2 levels) in cleaning'
     # ./vendor/bundle/ruby/2.5.0/gems/database_cleaner-1.8.5/lib/database_cleaner/configuration.rb:88:in `cleaning'
     # ./spec/spec_helper.rb:169:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.5.0/gems/webmock-3.12.2/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

```